src/USER-PLUMED/fix_plumed.cpp

```diff
@@ -0,0 +1,436 @@
+        //    it is defined inside plumed.
+        p->cmd("GREX setMPIIntercomm",&inter_comm);
+     }
+     p->cmd("GREX init",NULL);
+  }
+  // The general communicator is independent of the existence of partitions,
+  //   if there are partitions, world is defined within each partition,
+  //   whereas if partitions are not defined then world is equal to MPI_COMM_WORLD.
+  p->cmd("setMPIComm",&world);
+
+// Set up units
+// LAMMPS units wrt kj/mol - nm - ps
+// Set up units
+
+  if (force->boltz == 1.0){
+// LAMMPS units lj
+    p->cmd("setNaturalUnits");
+  } else {
+    double energyUnits=1.0;
+    double lengthUnits=1.0;
+    double timeUnits=1.0;
+    if (force->boltz == 0.0019872067){
+// LAMMPS units real :: kcal/mol; angstrom; fs
+      energyUnits=4.184;
+      lengthUnits=0.1;
+      timeUnits=0.001;
+    } else if (force->boltz == 8.617343e-5){
+// LAMMPS units metal :: eV; angstrom; ps
+      energyUnits=96.48530749925792;
+      lengthUnits=0.1;
+      timeUnits=1.0;
+    } else if (force->boltz == 1.3806504e-23){
+// LAMMPS units si :: Joule, m; s
+      energyUnits=0.001;
+      lengthUnits=1.e-9;
+      timeUnits=1.e-12;
+    } else if (force->boltz == 1.3806504e-16){
+// LAMMPS units cgs :: erg; cms;, s
+      energyUnits=6.0221418e13;
+      lengthUnits=1.e-7;
+      timeUnits=1.e-12;
+    } else if (force->boltz == 3.16681534e-6){
+// LAMMPS units electron :: Hartree, bohr, fs
+      energyUnits=2625.5257;
+      lengthUnits=0.052917725;
+      timeUnits=0.001;
+    } else error->all(FLERR,"Odd LAMMPS units, plumed cannot work with that");
+    p->cmd("setMDEnergyUnits",&energyUnits);
+    p->cmd("setMDLengthUnits",&lengthUnits);
+    p->cmd("setMDTimeUnits",&timeUnits);
+  }
+
+// Read fix parameters:
+  int next=0;
+  for(int i=3;i<narg;++i){
+    if(!strcmp(arg[i],"outfile")) next=1;
+    else if(next==1){
+      if(universe->existflag == 1){
+        // Each replica writes an independent log file
+        //  with suffix equal to the replica id
+        char str_num[32], logFile[1024];
+        sprintf(str_num,".%d",universe->iworld);
+        strncpy(logFile,arg[i],1024-32);
+        strcat(logFile,str_num);
+        p->cmd("setLogFile",logFile);
+        next=0;
+      } else {
+        // partition option not used
+        p->cmd("setLogFile",arg[i]);
+        next=0;
+      }
+    }
+    else if(!strcmp(arg[i],"plumedfile"))next=2;
+    else if(next==2){
+      p->cmd("setPlumedDat",arg[i]);
+      next=0;
+    }
+    else error->all(FLERR,"syntax error in fix plumed - use 'fix name plumed plumedfile plumed.dat outfile plumed.out' ");
+  }
+  if(next==1) error->all(FLERR,"missing argument for outfile option");
+  if(next==2) error->all(FLERR,"missing argument for plumedfile option");
+
+  p->cmd("setMDEngine","LAMMPS");
+
+  int natoms=int(atom->natoms);
+  p->cmd("setNatoms",&natoms);
+
+  double dt=update->dt;
+  p->cmd("setTimestep",&dt);
```

Comment:
please note, that certain features in LAMMPS may change the length of the timestep (`update->dt`), thus you should implement the `Fix::reset_dt()` API, even if only to throw an error that updating the timestep during a run is not supported by fix plumed.

Reply:
this is still an open issue